extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/standalone/VertxRequestHandler.java

```diff
@@ -114,6 +136,19 @@ private void dispatch(RoutingContext routingContext, InputStream is, VertxOutput
             if (!vertxRequest.getAsyncContext().isSuspended()) {
                 try {
                     vertxResponse.finish();
+                    // the reason we only cache successful responses is to ensure that a torrent of erroneous URLs
+                    // doesn't fill up the cache and cause a DoS
```

Comment:
I think you should cache unsuccesfull responses as well, especially if we go for having a proper cache.

DDoS are caused by the system being overwhelmed; hitting the "bad response cache" would be the best way to avoid it as the cache would help fighting load.

Reply:
I agree 💯%. If we have a proper cache than all this nastiness can go away.